src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ArrayConverter.cs

```diff
@@ -19,7 +18,6 @@ internal sealed class ArrayConverter<TCollection, TElement>
 
         protected override void Add(TElement value, ref ReadStack state)
         {
-            Debug.Assert(state.Current.ReturnValue is List<TElement>);
```

Comment:
These` Debug.Asserts` were conveying intent that the methods expect a particular type and a runtime invalid cast operation (due to the hard cast) is not expected. Why is that not useful to keep?